### PR TITLE
Remove WeakRefernce from MapboxMap and Style

### PR DESCRIPTION
**Pull request by Chaoba**
### Summary of changes
This pr removes the WeakRefernce form mapbox ans style, and throws `MapboxMapMemoryLeakException` when the nativeMap is null to notify the memory leak.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-kotlin-binary-compatibility` CI will fail.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.
 - [x] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ Mapbox welcomes participation and contributions from everyone.
 # main
 ## Bug fixes 🐞
 * Fix an issue when user subscribe sdk listeners multiple times, by changing CopyOnWriteArrayList to CopyOnWriteArraySet in the sdk to hold listeners. ([1183](https://github.com/mapbox/mapbox-maps-android/pull/1183))
+* Remove WeakRefernce from MapboxMap and Style. ([#1193](https://github.com/mapbox/mapbox-maps-android/pull/1193))
```
Comment (yunikkk) [inline CHANGELOG.md:8]:
```suggestion
* Remove WeakReference from MapboxMap and Style. ([#1193](https://github.com/mapbox/mapbox-maps-android/pull/1193))
```

Comment (yunikkk) [reply CHANGELOG.md:8]: But also a more general question - do we need to add a note about that at all? Seems it's an implementation detail and users should not know about it.

**Review by yunikkk — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Chaoba) [reply CHANGELOG.md:8]: Think about it. Yes, this is an internal change, no need to expose it to users.

**Review by yunikkk — APPROVED** [review]

**Comment by pengdev** [comment]
@Chaoba would it make sense to have a change log entry for this fix? as it introduced a new `MapboxMapMemoryLeakException`, and would be good to get 👁️ from @tobrun as well.

**Review by tobrun — APPROVED** [review]
Thank you for picking this up!

**Review by pengdev — APPROVED** [review]

**Pull request merged**